src/patterns/ec2-app/base.ts

```diff
@@ -377,6 +452,121 @@ export class GuEc2App extends Construct {
       }
     }
 
+    if (props.googleAuth?.enabled) {
+      const prefix = `/${scope.stage}/${scope.stack}/${app}`;
+
+      const {
+        allowedGroups = ["engineering@guardian.co.uk"],
+        credentialsSecretsManagerPath = `${prefix}/google-auth-credentials`,
+      } = props.googleAuth;
+
+      if (allowedGroups.length < 1) {
+        throw new Error("googleAuth.allowedGroups cannot be empty!");
+      }
+
+      if (allowedGroups.find((group) => !group.endsWith("@guardian.co.uk"))) {
+        throw new Error("googleAuth.allowedGroups must use the @guardian.co.uk domain.");
+      }
+
+      const deployToolsAccountId = StringParameter.fromStringParameterName(
+        scope,
+        "deploy-tools-account-id-parameter",
+        NAMED_SSM_PARAMETER_PATHS.DeployToolsAccountId.path
+      );
+
+      // See https://github.com/guardian/cognito-gatekeeper for the source code
+      // here.
+      // ARN format is: arn:aws:lambda:aws-region:acct-id:function:helloworld.
+      const gatekeeperFunctionArn = `arn:aws:lambda:eu-west-1:${deployToolsAccountId.stringValue}:function:deploy-PROD-gatekeeper-lambda`;
+
+      // Note, handler and filename must match here:
+      // https://github.com/guardian/cognito-gatekeeper.
+      const authLambda = new GuLambdaFunction(scope, "auth-lambda", {
+        app: app,
+        memorySize: 128,
+        handler: "devx-cognito-lambda-amd64-v1",
+        runtime: Runtime.GO_1_X,
+        fileName: "devx-cognito-lambda-amd64-v1.zip",
+        bucketNamePath: NAMED_SSM_PARAMETER_PATHS.OrganisationDistributionBucket.path,
+        architecture: Architecture.X86_64,
+        environment: {
+          ALLOWED_GROUPS: allowedGroups.join(","),
+          GATEKEEPER_FUNCTION_ARN: gatekeeperFunctionArn,
+        },
+      });
+
+      authLambda.addToRolePolicy(
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ["lambda:InvokeFunction"],
```

Comment:
Just double checking the singleton gateway lambda allows any org account to invoke it?

Reply:
I think you're right - we need a resource policy on the Gatekeeper lambda that allows invocation from our accounts. Will add it there.

Reply:
@akash1810 fixed this here! https://github.com/guardian/cognito-auth-lambdas/pull/3